### PR TITLE
linter: enable unconvert linter

### PR DESCRIPTION
**Pull request by vrothberg**
Detects unneccessary type conversions and helps in keeping the code base
cleaner.

Signed-off-by: Valentin Rothberg <vrothberg@redhat.com>

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: pkg/specgen/generate/container.go [hunk 1]
```diff
@@ -303,8 +303,8 @@ func FinishThrottleDevices(s *specgen.SpecGenerator) error {
 			if err := unix.Stat(k, &statT); err != nil {
 				return err
 			}
-			v.Major = (int64(unix.Major(uint64(statT.Rdev))))
```
Comment (Luap99) [inline pkg/specgen/generate/container.go:306]: I think we need this. Stat_t.Rdev is not uint64 on every architecture, compare https://github.com/containers/psgo/pull/77/files

Some problem with the other places. I am not sure if this allies to all fields in the stat struct but Rdev definitely one where we have to convert.

Comment (vrothberg) [reply pkg/specgen/generate/container.go:306]: Updated :heavy_check_mark:

**Comment by rhatdan** [comment]
LGTM

**Pull request merged**